eng/build.ps1

```diff
@@ -5,7 +5,7 @@ Param(
   [ValidateSet("Debug","Release","Checked")][string[]][Alias('c')]$configuration = @("Debug"),
   [string][Alias('f')]$framework,
   [string]$vs,
-  [ValidateSet("Windows_NT","Unix")][string]$os,
+  [ValidateSet("Windows_NT","Linux")][string]$os,
```

Comment:
What about OSX?

Reply:
Do we support cross-compiling for OSX?

Reply:
Why wouldn't we?  I should be able to build a managed library on any OS for any OS.

Reply:
Agree, but this switch is for the repository's root build in which managed libraries is only a subset of targets. I'm trying this out right now locally.

Reply:
We can for sure add OSX to the ValidateSet entries but I don't think that cross-building the runtime repo for OSX on Windows works today.

Reply:
Well, don't let this block fixing this and unblocking the source indexer.  But it's worth following up on.  It's a strange limitation that I can build all the managed libraries on Windows or Linux for Windows and Linux but not for OSX.  It's also strange that someone on OSX would get an error if they explicitly specified their own OS to target.

Reply:
> It's also strange that someone on OSX would get an error if they explicitly specified their own OS to target.

For historical reasons we assume that build.ps1 means running on Windows and build.sh means running on Unix. I know that @eiriktsarpalis hit issues with that in the past. That's an overall Arcade issue as it was designed with that assumption, cc @markwilkie.

What I'm saying is that a user on OSX will never hit this is limitation as it's only build.ps1 (Windows) specific.

> Well, don't let this block fixing this and unblocking the source indexer. But it's worth following up on.

Agreed. But realizing that the native build will never work, my argument that OSX wouldn't work doesn't hold as then Linux shouldn't be part of the allowed values either. Let me add the OSX value.